nav2_rrtconnect_planner/src/rrt_connect.cpp

```diff
@@ -0,0 +1,269 @@
+: tf_(nullptr), costmap_(nullptr), ss_(nullptr), ompl_state_space_(nullptr), bounds_(nullptr)
+{
+
+}
+
+void RRTConnect::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+{
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  ompl_initialized_ = false;
+
+  RCLCPP_INFO(
+    node_->get_logger(), "Configuring plugin %s of type NavfnPlanner",
+    name_.c_str());
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".solve_time",
+    rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + ".solve_time", solve_time_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".max_trials",
+    rclcpp::ParameterValue(4));
+  node_->get_parameter(name_ + ".max_trials", max_trials_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".collision_checking_resolution", rclcpp::ParameterValue(
+      0.001));
+  node_->get_parameter(name_ + ".collision_checking_resolution", collision_checking_resolution_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".allow_unknown", rclcpp::ParameterValue(
+      true));
+  node_->get_parameter(name_ + ".allow_unknown", allow_unknown_);
+}
+
+void RRTConnect::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::initializeOMPL()
+{
+  // Defining Bounds
+  bounds_ = new ompl::base::RealVectorBounds(2);
+  bounds_->setLow(0, costmap_->getOriginX());
+  bounds_->setHigh(
+    0, costmap_->getSizeInMetersX() +
+    costmap_->getOriginX());
+  bounds_->setLow(1, costmap_->getOriginY());
+  bounds_->setHigh(
+    1, costmap_->getSizeInMetersY() +
+    costmap_->getOriginY());
+
+  RCLCPP_INFO(
+    node_->get_logger(),
+    "bounds: %f %f %f %f", bounds_->low[0], bounds_->low[1],
+    bounds_->high[0], bounds_->high[1]);
+
+  ompl_state_space_ = std::make_shared<ompl::base::SE2StateSpace>();
+  // Bounding the R2 state space
+  ompl_state_space_->as<ompl::base::SE2StateSpace>()->setBounds(*bounds_);
+
+  // Simple setup
+  ss_.reset(new ompl::geometric::SimpleSetup(ompl_state_space_));
+  // Setting state validity checker
+  ss_->setStateValidityChecker(
+    [this](const ompl::base::State * state) {
+      return this->isStateValid(state);
+    });
+  // Setting the state validity checking resolution
+  ss_->getSpaceInformation()->setStateValidityCheckingResolution(collision_checking_resolution_);
```

Comment:
`collision_checking_resolution_`, I'm not sure off hand what that means - can this be the costmap resolution?

Reply:
This resolution is for the state validator. It means how deep the validator has to go if checking for obstacles between two sampled states. If you pass costmap resolution it doesn't work nice.

Reply:
If 0.05 is "typical", I'd think 0.025 to be reasonable (2 sample potentially per pixel). Or am I missing something?

Reply:
0.025 also doesn't work. But 0.0125 does work. I am not sure but I think it some thing related to interpolation which `ompl::base::DiscreteMotionValidator` uses for checking obstacles between two states.

Reply:
Mhm, maybe something to we should try to understand more. "How" didnt it work? Do you know what the units for this are? Is it configuration space units (so for now the costmap size, so meters I believe from yoru setup)? Or interpolated normalized? 

As you can tell, its been awhile since I looked at OMPL code, apologies. I should know this better.

Reply:
No worries! It's okay not to know :smiley:. This is my first time working with OMPL. I would dig deeper and try to find out about the problem.